web/html/src/manager/systems/bootstrap/bootstrap-minions.js

```diff
@@ -198,11 +246,43 @@ class BootstrapMinions extends React.Component {
                     </div>
                 </div>
                 <div className="form-group">
+                    <label className="col-md-3 control-label">Authentication method:</label>
+
+                    <div className="col-md-6">
+                        <div className="radio col-md-3">
+                            <label>
+                                <input name="authMethod" type="radio" value="password" checked={this.state.authMethod === "password"} onChange={this.authMethodChanged}/>
+                                <span>{t("Password")}</span>
+                            </label>
+                        </div>
+                        <div className="radio col-md-3">
+                            <label>
+                                <input name="authMethod" type="radio" value="ssh-key" checked={this.state.authMethod === "ssh-key"} onChange={this.authMethodChanged}/>
+                                <span>{t("SSH Private Key")}</span>
+                            </label>
+                        </div>
+                    </div>
+                </div>
+                <div className="form-group" style={this.state.authMethod === "password" ? {} : {display: "none"} } >
```

Comment:
What about creating a conditional HTML content variable before the `return` and just showing it here instead?

Something like:
```js
const conditionalAuthContent = this.state.authMethod === "password" ?
    <div>
      // the password input box
    </div>
    :
    <div>
      // the  ssh-key box content (privKeyFile and privKeyPwd inputs)
    </div>
```

and then in the return body just use `{conditionalAuthContent}`.

Reply:
I did that in the original version, but for some reason React was complaining about changing controlled elements to uncontrolled and vice versa, but this is not the case anymore. I've changed as your suggested.